aten/src/ATen/native/Unique.cpp

```diff
@@ -154,6 +154,28 @@ std::tuple<Tensor, Tensor, Tensor> _unique_dim_cpu_template(
     const bool consecutive,
     const bool return_inverse,
     const bool return_counts) {
+  // check if dim is in range and see if the size along that dim is 0
+  if (dim <= self.dim() && self.size(dim) == 0) {
```

Comment:
What would happen if `self.size(dim) != 0` but there is another dim that has zero size? Shouldn't this case be handled as well?